### PR TITLE
Changes to energy units and offset do not affect .calc.energy

### LINKED ISSUE #86: BUG: changed energy units, got same energy value
When developing unit tests for new `hkl.diffract.Diffractometer.wh()` method (#78) discovered that reported energy value did not change when the units were changed from `keV` to `eV`.

```python
def test_change_energy_units(fourc):
    assert fourc.energy.get() == 8
    assert fourc.energy_units.get() == "keV"
    fourc.energy_units.put("eV")
    assert fourc.energy.get() == 8000
```

results in a test failure:

```
========================================================= FAILURES =========================================================
_________________________________________________ test_change_energy_units _________________________________________________

fourc = Fourc(prefix='', name='fourc', settle_time=0.0, timeout=None, egu='', limits=(0, 0), source='computed', read_attrs=['h...ck', 'l.setpoint', 'omega', 'chi', 'phi', 'tth'], configuration_attrs=['energy', 'UB', 'h', 'k', 'l'], concurrent=True)

    def test_change_energy_units(fourc):
        assert fourc.energy.get() == 8
        assert fourc.energy_units.get() == "keV"
        fourc.energy_units.put("eV")
>       assert fourc.energy.get() == 8000
E       assert 8.0 == 8000
E         +8.0
E         -8000

hkl/tests/test_diffract.py:171: AssertionError
================================================= short test summary info ==================================================
FAILED hkl/tests/test_diffract.py::test_change_energy_units - assert 8.0 == 8000
=============================================== 1 failed, 6 passed in 1.96s ================================================
```

### PR DESCRIPTION
**Pull request by prjemian**
fixes #86

Changes to units or offset will only change `.energy`, not `.calc.energy`.  To change `.calc.energy`, must set `.energy`.

To change energy, offset, & units, do it in this order:

1. set units
2. set offset
3. set energy

### CODE REVIEW

**Comment by prjemian** [comment]
@mrakitin ready for review

**Comment by mrakitin** [comment]
There is a conflicting file here. Will review once it's resolved.

**Comment by prjemian** [comment]
@mrakitin Merge conflict is resolved.  Ready to review.

File: hkl/diffract.py [hunk 1]
```diff
@@ -223,9 +244,59 @@ def _energy_changed(self, value=None, **kwargs):
...
+        # TODO: is there a loop back through _update_calc_energy?
+        # raise NotImplementedError(f"units = {self.energy_units.get()}")
+        try:
+            energy = pint.Quantity(
+                self.calc.energy, "keV"
+            ).to(
+                self.energy_units.get()
+            )
+        except Exception as exc:
+            raise NotImplementedError(f"units = {self.energy_units.get()}: {exc}")
```
Comment (mrakitin) [inline hkl/diffract.py:273]: `self.energy_units.get()` is called twice. An optional suggestion: stash it before the try..except block, and reuse.

**Review by mrakitin — APPROVED** [review]
Looks good! Have a minor, non-blocking suggestion.

**Comment by prjemian** [comment]
Synchronize instructions to set the energy with https://github.com/bluesky/hklpy/pull/101#discussion_r553482099 before merging.

**Pull request merged**